sphinx/source/docs/user_guide/tools.rst

```diff
@@ -561,6 +562,218 @@ shown below:
 .. bokeh-plot:: docs/user_guide/examples/tools_hover_custom_tooltip.py
     :source-position: above
 
+
+.. _userguide_tools_edit:
+
+Edit Tools
+----------
+
+The edit tools provide functionality for drawing and editing glyphs
+client-side by adding, modifying and deleting ``ColumnDataSource``
+data.
+
+All the edit tools share a small number of key bindings:
+
+* SHIFT: Allows adding to a selection or continue drawing
+* DELETE: Deletes the selected glyphs
+* ESC: Clear the selection
+
+.. note::
+
+   On some shortened keyboards (including MacBooks) the DELETE key is
+   missing and can be toggled by pressing FN + BACKSPACE
+
+BoxDrawTool
+~~~~~~~~~~~
+
+* name: ``'box_draw'``
+* menu icon: |box_draw_icon|
+
+The BoxDrawTool allows drawing, dragging and deleting ``Rect`` glyphs
+on one or more renderers by editing the underlying
+``ColumnDataSource`` data. Like other drawing tools, the renderers
+that are to be edited must be supplied explicitly as a list. When
+drawing a new box the data will always be added to the
+``ColumnDataSource`` on the first supplied renderer.
```

Comment:
it's never demonstrated anywhere what "supplying as a list" looks like. This needs an example code snippet to show what configuring one of these tools looks like in some way. Additionally there should at least be a link to the code for the live demos. If it is too cumbersome to put directly inline, then a "The code for this example is at..." would do. Or there is also a collapsible code block directive to make it hidden by default.

Reply:
I think the PolyEdit example does demonstrate it. Should I just modify each example to use multiple renderers?

>Or there is also a collapsible code block directive to make it hidden by default.

I'll give that a go.

Reply:
No I think the example is fine, the snippet does not have to relate to the example. Just a few lines showing e.g.
```
r1 = plot.rect(...)
r2 = plot.rect(...)

plot.add_tool(BoxDrawTool(renderers=(r1, r2))
```
slightly more fleshed out

Reply:
Sounds good, should I do this for each tool?